src/core/thread/mesh_forwarder.cpp

```diff
@@ -573,12 +574,32 @@ void MeshForwarder::SetRxOnWhenIdle(bool aRxOnWhenIdle)
 
 void MeshForwarder::SetPollPeriod(uint32_t aPeriod)
 {
-    if (mMac.GetRxOnWhenIdle() == false && mPollPeriod != aPeriod)
+    if (!mNetif.IsUp())
```

Comment:
I think `SetPollPeriod` should only be called by a client of OpenThread as part of configuration.  Then the actual poll period used within OpenThread is the minimum of:

1. the configured poll period or
2. the Child Timeout value minus a delta